FieldOpt/Reservoir/grid/cell.cpp

```diff
@@ -52,24 +53,35 @@ namespace Reservoir {
         bool Cell::EnvelopsPoint(Eigen::Vector3d point) {
             bool point_inside = true;
             for (Face face : faces_) {
-                if ((point - face.corners[0]).dot(face.normal_vector) < 0)
+		double dot_prod = (point - face.corners[0]).dot(face.normal_vector);
+		if ( dot_prod < 0)
                     point_inside = false;
             }
             return point_inside;
         }
 
         void Cell::initializeFaces() {
+//            int face_indices_points[6][4] = {
+//                     {0, 2, 1, 3},
+//                     {4, 5, 6, 7},
+//                     {0, 4, 2, 6},
+//                     {1, 3, 5, 7},
+//                     {0, 1, 4, 5},
+//                     {2, 6, 3, 7}
+//            };
+//
+
             int face_indices_points[6][4] = {
-                    {0, 2, 1, 3},
-                    {4, 5, 6, 7},
-                    {0, 4, 2, 6},
-                    {1, 3, 5, 7},
-                    {0, 1, 4, 5},
-                    {2, 6, 3, 7}
+                    {2, 0, 3, 1},
```

Comment:
Aha, then I get it! I'll just keep the old definition for now then (with your commented out), just to keep the tests working for now.

Looks like I'll have both your pull requests merged by the end of today. How do you want me to do it? (ref: the email I sent you yesterday).